test/provisioner/provision_azure.go

```diff
@@ -8,37 +8,355 @@ package provisioner
+
 	return nil
 }
 
-func (l *AzureCloudProvisioner) DeleteVPC(ctx context.Context, cfg *envconf.Config) error {
+func (p *AzureCloudProvisioner) DeleteCluster(ctx context.Context, cfg *envconf.Config) error {
+	log.Trace("DeleteCluster()")
+	log.Infof("Deleting Cluster %s.\n", AzureProps.ClusterName)
+	_, err := AzureProps.ManagedaksClient.Delete(context.Background(), AzureProps.ResourceGroupName, AzureProps.ClusterName)
+	if err != nil {
+		return err
+	}
+
+	clusterRemoved := false
+	waitMinutes := 10
+	log.Infof("Waiting for cluster %s to be removed...\n", AzureProps.ClusterName)
+	for i := 0; i <= waitMinutes; i++ {
+		found, err := clusterExists(AzureProps.ResourceGroupName, AzureProps.ClusterName)
+		if err != nil {
+			log.Warnf("Err %s happened when retrieve cluster, try again...\n", err)
+			continue
+		}
+		if !found {
+			log.Infof("Cluster %s is removed.\n", AzureProps.ClusterName)
+			clusterRemoved = true
+			break
+		}
+		time.Sleep(60 * time.Second)
+	}
+
+	if !clusterRemoved {
+		return fmt.Errorf("Cluster %s was not removed completely in %d minutes.\n", AzureProps.ClusterName, waitMinutes)
+	}
+
 	return nil
 }
 
-func (l *AzureCloudProvisioner) UploadPodvm(imagePath string, ctx context.Context, cfg *envconf.Config) error {
+//func (p *AzureCloudProvisioner) DoKustomize(ctx context.Context, cfg *envconf.Config) error {
+//	overlayFile := "../../install/overlays/azure/kustomization.yaml"
+//	overlayFileBak := "../../install/overlays/azure/kustomization.yaml.bak"
+//	//runtimeCri := "imagepullsecret.yaml"
+//	err := os.Rename(overlayFile, overlayFileBak)
+//	if err != nil {
+//		return err
+//	}
+//
+//	input, err := os.ReadFile(overlayFileBak)
+//	if err != nil {
+//		return err
+//	}
+//
+//	replacer := strings.NewReplacer("CLOUD_PROVIDER=\"\"", "CLOUD_PROVIDER=\""+AzureProps.CloudProvider+"\"",
+//		"AZURE_SUBSCRIPTION_ID=\"\"", "AZURE_SUBSCRIPTION_ID=\""+AzureProps.SubscriptionID+"\"",
+//		"AZURE_REGION=\"\"", "AZURE_REGION=\""+AzureProps.Location+"\"",
+//		"AZURE_INSTANCE_SIZE=\"\"", "AZURE_INSTANCE_SIZE=\""+AzureProps.InstanceSize+"\"",
+//		"AZURE_SUBNET_ID=\"\"", "AZURE_SUBNET_ID=\""+AzureProps.SubnetID+"\"",
+//		"AZURE_RESOURCE_GROUP=\"\"", "AZURE_RESOURCE_GROUP=\""+AzureProps.ResourceGroupName+"\"",
+//		"AZURE_CLIENT_ID=\"\"", "AZURE_CLIENT_ID=\""+AzureProps.ClientID+"\"",
+//		"AZURE_CLIENT_SECRET=\"\"", "AZURE_CLIENT_SECRET=\""+AzureProps.ClientSecret+"\"",
+//		"AZURE_TENANT_ID=\"\"", "AZURE_TENANT_ID=\""+AzureProps.TenantID+"\"",
+//		"AZURE_IMAGE_ID=\"\"", "AZURE_IMAGE_ID=\""+AzureProps.ImageID+"\"")
+//
+//	output := replacer.Replace(string(input))
+//
+//	if err = os.WriteFile(overlayFile, []byte(output), 0666); err != nil {
+//		return err
+//	}
+//
+//	return nil
+//}
+
+func (p *AzureCloudProvisioner) UploadPodvm(imagePath string, ctx context.Context, cfg *envconf.Config) error {
+	log.Trace("UploadPodvm()")
+	log.Trace("Image is uploaded via packer in case of azure")
```

Comment:
Where do we trigger packer in this case?

Reply:
I will handle that in next PR where will get whole e2e test working along with Wainer's PR on install overlays.